### DISCUSSION TITLE
Load collection and insert data

### DISCUSSION THREAD

**Discussion by dod-ia** [milvus-io/milvus#10935] [Q&A and General discussion] [answered]
During an "insert campaign", i would like verify (for each new data) if the data is already indexed in the database by checking its ID (which it is defined by me) else ignore it. For each step, I must be sure there is not duplication so the database must be up-to-date at each insertion step. To check if the ID is already indexed, I have to realize a query and, to do that, load the collection in memory before. The load operation is time-consuming and not very efficient for massive insertion. I use Milvus 2.0.
What's the good approach to do that in an efficient way ? The query/load step is necessary or there is an another method ?

Thanks for your help !

Edit: Resolved: Misunderstanding with the API - Dideplucation/Search by id in the Roadmap

**Comment by xiaofan-luan** [comment, marked as answer]
You only need to load once and then you query by id would work even if insertion happens after load.
Primary key deduplication is already in our roadmap,  once the functionality is ready, you no longer need to do check by yourself~

**Reply by martinenkoEduard** [reply]
Why do you even need to load a collection? 
Won't it work without .load() command?

Can I insert new values after .load() command?

**Reply by xiaofan-luan** [reply]
> Why do you even need to load a collection? Won't it work without .load() command?
> 
> Can I insert new values after .load() command?

Yes, you can insert after load. 
Since milvus put all data in memory, and some users don't have enough memory. 
So you always need to load once before you search, but you can insert without load

**Reply by LMERose** [reply]
hello, if I want to use this :create_collection --> insert（） --> create_index（） --> load（） --> search（）,what should I do?Which part of the program should I modify? I'm using the langchain interface

**Reply by barzan-hayati** [reply]
Does this resolved in the newest version of milvus?
> Primary key deduplication is already in our roadmap

**Reply by liliu-z** [reply]
> Does this resolved in the newest version of milvus?
> 
> > Primary key deduplication is already in our roadmap

[Upsert](https://milvus.io/docs/insert-update-delete.md#Upsert-entities) can help on this

**Reply by liliu-z** [reply]
> hello, if I want to use this :create_collection --> insert（） --> create_index（） --> load（） --> search（）,what should I do?Which part of the program should I modify? I'm using the langchain interface

Did meet any issue with this order?
